app/(shared)/public/conversations/[conversationId]/messages/route.ts

```diff
@@ -0,0 +1,29 @@
+import { and, asc, eq } from "drizzle-orm";
+import { NextRequest, NextResponse } from "next/server";
+import { ZodError } from "zod";
+
+import { conversationMessagesResponseSchema } from "@/lib/api";
+import db from "@/lib/server/db";
+import * as schema from "@/lib/server/db/schema";
+
+export async function POST(request: NextRequest, { params }: { params: Promise<{ conversationId: string }> }) {
```

Comment:
This is strange as a POST request.  If you need the tenant it should be passed in the URL.  I would also see if I can scope the tenant by slug for consistency.  You technically don't need the tenant here though because conversation IDs are UUIDs.